crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -222,14 +221,26 @@ pub(crate) fn type_check_expression(
             )
         }
         HirExpression::Block(block_expr) => {
-            for stmt in block_expr.statements() {
-                super::stmt::type_check(interner, stmt, errors);
-            }
+            let mut last = None;
```

Comment:
Can we make this the unit Type instead of an optional?

Reply:
Good idea